sdk/src/main/java/com/mapbox/maps/MapView.kt

```diff
@@ -227,7 +230,21 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
    *
    * @return [MapboxMap] object to interact with the map.
    */
-  override fun getMapboxMap(): MapboxMap = mapController.getMapboxMap()
+  override fun getMapboxMap(): MapboxMap {
+    val activity = context.getActivity()
+    if (activity == null || activity.isFinishing || activity.isDestroyed) {
+      if (viewAnnotationManagerDelegate.isInitialized()) {
+        (viewAnnotationManager as ViewAnnotationManagerImpl).destroy()
+      }
+      mapController.onDestroy()
```

Comment:
just call `onDestroy()` here from `MapView`? no need to duplicate the code

Reply:
Didn't want to call `onDestroy` just in case anything in the superclass is affected, but agree it would be better to extract this functionality to its own method.